### ISSUE TITLE
Debian complains about old OpenPGP signatures

### ISSUE THREAD

**Issue by anarcat** [a-j-wood/pv#71]
Hi again! :)

I have tried to ship your OpenPGP signature attached to the 1.7.0 release, and this is what our build scripts said about it:

```
anarcat@angela:pkg-pv$ git bp 
gbp:info: All Orig tarballs 'pv_1.7.0.orig.tar.bz2' found at '..'
gbp:info: Extracting 'pv_1.7.0.orig.tar.bz2' to '/home/anarcat/src/build-area/pv-tmp'
gbp:info: Exporting 'HEAD' to '/home/anarcat/src/build-area/pv-tmp'
gbp:info: Moving '/home/anarcat/src/build-area/pv-tmp' to '/home/anarcat/src/build-area/pv-1.7.0'
gbp:info: Performing the build
dpkg-source: info: using source format '3.0 (quilt)'
dpkg-source: info: verifying ./pv_1.7.0.orig.tar.bz2.asc
Error: Policy rejected packet type

Caused by:
    Signature Packet v3 is not considered secure since 2007-02-01T00:00:00Z
dpkg-source: warning: cannot verify upstream tarball signature for ./pv_1.7.0.orig.tar.bz2: no acceptable signature found
[...]
```

I must admit I'm not very familiar with this, the latest (draft) RFC mention v3 packets are deprecated: https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-10.html#name-public-key-packet-formats

... but that doesn't really say how to fix this issue. I suspect you might need to generate a new OpenPGP key, given that the one I have on file is a DSA1024 key, deprecated, see also https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-10.html#name-dsa

for now I just won't ship the signature alongside the debian package, but it would be nice to see that key updated...

thanks!

**Comment by a-j-wood** [comment]
The signature was created with GnuPG 2.0.14 on CentOS 6, using a key I generated in December 2001, so I have some modernising to do.

**Comment by a-j-wood** [comment]
I think I'm going to have to just generate a new key and use that in future.  I don't distribute it anywhere, it just sits on my web site - is there anything specific I should do regarding key distribution, or do you think just replacing the one I link to on my web site will be sufficient?

**Comment by a-j-wood** [comment]
I've now generated a new key and re-signed the package.  Hopefully this will work better.

**Comment by anarcat** [comment]
When I do a key transition, I typically write a signed document
explaining what i did and what to do with the change.

I also sign the new key with my old key.

I transitioned recently to an ECC key, and issued the following
statement for example:

https://anarc.at/openpgp-transition-2023.txt

I also wrote some documentation on that process here:

https://gitlab.torproject.org/tpo/tpa/team/-/wikis/howto/openpgp#rotating-keys

HTH!
